onnx/defs/shape_inference.h

```diff
@@ -13,6 +13,60 @@ namespace ONNX_NAMESPACE {
 
 using Dim = TensorShapeProto_Dimension;
 
+struct SymbolTable {
+  SymbolTable(GraphProto& g) : index_(0){
+    addFromGraph(g);
+  }
+  void addFromGraph(GraphProto& g) {
```

Comment:
nit: add blank line before mtd start and after mtd end

Reply:
nit: do you need addFromGraph? can you directly call AddExistingSymbolicDims from the constructor?

Reply:
addFromGraph has another purpose. If shape_inference bumps into a sub-graph, it will rather use addFromGraph to add symbols from sub graph.

Reply:
Technically, we need to decide on the scope of the dimension-symbols like M and N. The existing spec is
```
Currently, dimension variables are not scoped. A dimension variable "N" represents the same value
across the entire graph in a model. For example, if the graph has two inputs X and Y each with
shape ["N"], then at runtime the values passed in for X and Y MUST be tensors of rank 1 with the
same dimension. Nested sub-graphs currently share the same scope for dimension variables as the
main-graph. This allows a model to relate the dimensions of tensors inside the subgraph to the
dimensions of tensors in the outer graph.
```
(see under https://github.com/gramalingam/onnx/blob/master/docs/IR.md#static-tensor-shapes ). If we stick to this, we should visit all subgraphs and add their symbols right when constructing the top model-level symbol table.